tests/unit/mainwindow/test_prompt.py

```diff
@@ -130,3 +131,14 @@ def test_root_path(self, get_prompt):
         """With / as path, show root contents."""
         prompt = get_prompt('/')
         assert prompt._file_model.rootPath() == '/'
+    
+    def test_expand_user(self, qtbot, get_prompt):
+        prompt = get_prompt('/')
+        prompt._lineedit.setText('')
+        qtbot.keyPress(prompt._lineedit, Qt.Key_AsciiTilde)
+        # The path currently seen by the file_model should be the same as os.path.expanduser('~')
+        assert str(Path(prompt._file_model.rootPath()) / prompt._to_complete) == os.path.expanduser('~')
```

Comment:
You could compare the pathlib object directly:

```suggestion
        assert Path(prompt._file_model.rootPath()) / prompt._to_complete == pathlib.Path.home()
```